caustic/sims/lens_source.py

```diff
@@ -0,0 +1,121 @@
+from .simulator import Simulator
+from torch.nn.functional import avg_pool2d, conv2d
+from typing import Tuple, Optional
+import torch
+
+__all__ = ("Lens_Source",)
+
+class Lens_Source(Simulator):
+    """Lens image of a source.
+    
+    Striaghtforward simulator to sample a lensed image of a source
+    object. Constructs a sampling grid internally based on the
+    pixelscale and gridding parameters. It can automatically upscale
+    and fine sample an image. This is the most straightforward
+    simulator to view the image if you aready have a lens and source
+    chosen.
+
+    Example usage::
+
+       import matplotlib.pyplot as plt
+       import caustic
+
+       cosmo = caustic.FlatLambdaCDM()
+       lens = caustic.lenses.SIS(cosmology = cosmo, x0 = 0., y0 = 0., th_ein = 1.)
+       source = caustic.sources.Sersic(x0 = 0., y0 = 0., q = 0.5, phi = 0.4, n = 2., Re = 1., Ie = 1.)
+       sim = caustic.sims.Lens_Source(lens, source, pixelscale = 0.05, gridx = 100, gridy = 100, upsample_factor = 2, z_s = 1.)
+       
+       img = sim()
+       plt.imshow(img, origin = "lower")
+       plt.show()
+
+    Attributes:
+      lens_mass: caustic lens mass model object
+      source_light: caustic light object which defines the background source
+      lens_light: caustic light object which defines the lensing object's light
+      psf: An image to convolve with the scene. Note that if ``upsample_factor > 1`` the psf must also be at the higher resolution.
+      pixelscale: pixelscale of the sampling grid.
+      gridx: number of pixels on the x-axis for the sampling grid
+      gridy: number of pixels on the y-axis for the sampling grid. If left as ``None`` then this will simply be equal to ``gridx``
+      upsample_factor: Amount of upsampling to model the image. For example ``upsample_factor = 2`` indicates that the image will be sampled at double the resolution then summed back to the original resolution (given by pixelscale and gridx/y).
+      psf_pad: If convolving the PSF it is important to sample the model in a larger FOV equal to half the PSF size in order to account for light that scatters from outside the requested FOV inwards. Internally this padding will be added before sampling, then cropped off before returning the final image to the user.
+      z_s: redshift of the source
+      name: a name for this simulator in the parameter DAG.
+
+    """
+    def __init__(
+        self,
+        lens_mass,
+        source_light,
+        lens_light = None,
+        psf = None,
+        pixelscale: float = 0.05,
```

Comment:
I think for this PR we can just keep it in line with `Pixelated` and `PixelatedConvergence` I have been meaning to go through caustic and clean up variable names. For example in the SIE we are still using `b` for the einstein radius even though everywhere else its `th_ein` and really I think it should be `Rein` since that's more clear to me. So I agree with `pixelscale` to `pixel_scale` but just not in this PR